apps/src/levelbuilder/lesson-editor/UploadImageEditor.jsx

```diff
@@ -0,0 +1,105 @@
+import ImageEditor from '@toast-ui/react-image-editor';
+import PropTypes from 'prop-types';
+import React, {useRef} from 'react';
+
+import 'tui-image-editor/dist/tui-image-editor.css';
+import Button from '@cdo/apps/legacySharedComponents/Button';
+
+import LessonEditorDialog from './LessonEditorDialog';
+
+import styles from './uploadImage.module.scss';
+
+const editorOptions = {
+  includeUI: {
+    theme: {},
+    menu: ['crop', 'flip', 'rotate', 'draw', 'shape', 'text', 'resize'],
+    uiSize: {
+      width: '100%',
+      height: '100%',
+    },
+    menuBarPosition: 'left',
+  },
+  cssMaxWidth: 700,
+  cssMaxHeight: 500,
+  selectionStyle: {
+    cornerSize: 20,
+    rotatingPointOffset: 70,
+  },
+};
+
+const dialogStyles = {
+  height: '80vh',
+};
+
+export default function UploadImageEditor({
+  imageUrl,
+  onSave,
+  onCancel,
+  isOpen,
+}) {
+  const editorRef = useRef(null);
+
+  const handleSave = () => {
+    const editorInstance = editorRef.current.getInstance();
+    const editedImageUrl = editorInstance.toDataURL();
+
+    // Convert Data URL to a File object
+    const dataUrlToFile = (dataUrl, filename) => {
+      const arr = dataUrl.split(',');
+      const mime = arr[0].match(/:(.*?);/)[1];
+      const bstr = atob(arr[1]);
+      let n = bstr.length;
+      const u8arr = new Uint8Array(n);
+      while (n--) {
+        u8arr[n] = bstr.charCodeAt(n);
+      }
```

Comment:
Was this chatGPT generated?  I had to google some of the functions and variable names to understand what was going on here.  I like the comment at the top about what this does, BUT, if possible, I wonder if we could add in some citation about where this comes from.